torch/ao/quantization/fx/_decomposed.py

```diff
@@ -122,3 +123,27 @@ def quantize_per_channel(input, scales, zero_points, axis, quant_min, quant_max,
 
     out = res.permute(tuple(permute_axis_list))
     return out.to(dtype)
+
+# Note: quant_min/quant_max/dtype are not used in the operator, but for now it's kept in
+# the signature as metadata for the input Tensor, this might be useful for pattern
+# matching in the future
+# We will revisit this later if we found there are no use cases for it
+quantized_decomposed_lib.define(
+    "dequantize_per_channel(Tensor input, Tensor scales, Tensor zero_points, int axis, int quant_min, int quant_max, ScalarType dtype) -> Tensor")
+
+@impl(quantized_decomposed_lib, "dequantize_per_channel", "CompositeExplicitAutograd")
+def dequantize_per_channel(input, scales, zero_points, axis, quant_min, quant_max, dtype):
+    assert input.dtype == dtype, f"Expecting input to have dtype torch.float32, but got dtype: {input.dtype}"
+    assert axis < input.dim(), f"Expecting axis to be < {input.dim()}"
+    _quant_min_max_bounds_check(quant_min, quant_max, dtype)
+    input, permute_axis_list = _permute_to_axis_zero(input, axis)
+    res = torch.zeros_like(input, dtype=torch.float32)
+
+    for i in range(input.size(0)):
+        # TODO: investigate why
+        # (input[i] - zero_points[i]).to(torch.float32) * scales[i]
+        # failed the test
+        res[i] = (input[i].to(torch.float32) - zero_points[i]) * scales[i]
```

Comment:
I think it's pretty important for all the quantize/dequantize functions of all flavors to be doing exactly the same thing, regardless of whether they are using quantized tensor or the decomposed representation.

What do you think about having this line of code implemented in c++ with the decomposed signature, and having that used everywhere?

Reply:
yeah I think it makes sense, and ideally fake quant should use the same so that it's exactly matching when people use QAT as well

Reply:
opened an issue in https://github.com/pytorch/pytorch/issues/89303 to track this